sed/core/processor.py

```diff
@@ -1483,6 +1489,129 @@ def calibrate_delay_axis(
+        rename: str = None,
+    ) -> None:
+        """Shift the delay axis of the dataframe by a constant or other columns.
+
+        Args:
+            constant (float, optional): The constant to shift the delay axis by.
+            columns (Union[str, Sequence[str]]): Name of the column(s) to apply the shift from.
+            weights (Union[float, Sequence[float]]): Weight of the shift to apply. A positive
+                sign shifts the delay axis to higher values. Defaults to +1.
+            preserve_mean (bool): Whether to subtract the mean of the column before applying the
+                shift. Defaults to False.
+            reductions (str): The reduction to apply to the column. Should be an available method
+                of dask.dataframe.Series. For example "mean". In this case the function is applied
+                to the column to generate a single value for the whole dataset. If None, the shift
+                is applied per-dataframe-row. Defaults to None. Currently only "mean" is supported.
+            inplace: Whether to apply the shift inplace or generate a new column.
+            rename: Whether to rename the column to "delay" after the shift.
+
+        Returns:
+            None
+        """
+        print("Adding delay offset to dataframe:")
+        delay_column = self._config["dataframe"]["delay_column"]
+        if delay_column not in self._dataframe.columns:
+            raise ValueError(
+                f"Energy column {delay_column} not found in dataframe! "
+                "Run `append energy axis` first.",
+            )
+
+        if self.dataframe is not None:
+            df, metadata = self.dc.add_offsets(
+                df=self._dataframe,
+                constant=constant,
+                flip_delay_axis=flip_delay_axis,
+                columns=columns,
+                delay_column=delay_column,
+                weights=weights,
+                reductions=reductions,
+                preserve_mean=preserve_mean,
+                inplace=inplace,
+                rename=rename,
+            )
+        if self._timed_dataframe is not None:
+            if delay_column in self._timed_dataframe.columns:
+                tdf, _ = self.dc.add_offsets(
+                    df=self._timed_dataframe,
+                    constant=constant,
+                    flip_delay_axis=flip_delay_axis,
+                    columns=columns,
+                    delay_column=delay_column,
+                    weights=weights,
+                    reductions=reductions,
+                    preserve_mean=preserve_mean,
+                    inplace=inplace,
+                    rename=rename,
+                )
+            self._attributes.add(
+                metadata,
+                "add_delay_offset",
+                duplicate_policy="append",
+            )
+            self._dataframe = df
+            if self._timed_dataframe is not None:
+                self._timed_dataframe = tdf
```

Comment:
This can be undefined, if the column does not exist! Rather, assign it directly as done in the other functions...

Reply:
This structure allows to use metadata duplicate_policy= raise correctly, without first overwriting the dataframe.

added a check to avoid the case when the column does not exist.